prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -409,6 +427,18 @@ func (o *KubernetesOptions) BuildClusterManagers(dryRun bool, callBack func(), o
 	return res, aggregatedErr
 }
 
+// checkPodListability checks if we are able to list pods using the provided
+// build cluster configuration.
+func checkPodListability(config rest.Config, namespace string) error {
+	client, err := ctrlruntimeclient.New(&config, ctrlruntimeclient.Options{})
+	if err != nil {
+		return err
+	}
+
+	var pods k8scorev1.PodList
+	return client.List(context.TODO(), &pods, ctrlruntimeclient.MatchingLabels{kube.CreatedByProw: "true"}, ctrlruntimeclient.InNamespace(namespace), ctrlruntimeclient.Limit(1))
```

Comment:
This check isn't sufficient for `prow-controller-manager`. It actually needs permission to create and get pods as well.  (This might be why prow-controller-manager crashlooped when other components that have build cluster clients didn't?)

Rather than actually doing a list/create/get can we instead use the equivalent of `kubectl auth can-i ...`? Listing pods could delay startup significantly and creating a pod would be wasteful and require cleanup.

Reply:
> This check isn't sufficient for `prow-controller-manager`.

True, but it would probably be enough to get PCM past the error that we saw about being unable to list pods.

> It actually needs permission to create and get pods as well.

True --- it would be more robust if we check for those permissions also. I'll compare against the RBAC rules we define for PCM in our integration tests.

> (This might be why prow-controller-manager crashlooped when other components that have build cluster clients didn't?)

Yes, I think so too.

> Rather than actually doing a list/create/get can we instead use the equivalent of `kubectl auth can-i ...`? Listing pods could delay startup significantly and creating a pod would be wasteful and require cleanup.

I didn't know about `kubectl auth can-i`! Definitely sounds like the perfect use case here. I will try to lift that underlying code from upstream and use it here.

Reply:
FTR, these lines were stolen from https://github.com/kubernetes/test-infra/blob/e57429125b2e2617bdd1fc2db00420499e1710a1/prow/plank/reconciler.go#L230-L233. I suppose we should update that location as well with a `kubectl auth can-i` equivalent? Maybe not in this PR but as a follow-up.